platform/windows/os_windows.cpp

```diff
@@ -1324,13 +1324,16 @@ void OS_Windows::unset_environment(const String &p_var) const {
 }
 
 String OS_Windows::get_stdin_string() {
-	WCHAR buff[1024];
-	DWORD count = 0;
-	if (ReadConsoleW(GetStdHandle(STD_INPUT_HANDLE), buff, 1024, &count, nullptr)) {
-		return String::utf16((const char16_t *)buff, count);
-	}
-
-	return String();
+    LocalVector<char> buff;
+    DWORD size = 0, count = 0;
+    HANDLE pipe = GetStdHandle(STD_INPUT_HANDLE);
+    if (PeekNamedPipe(pipe, nullptr, 0, nullptr, &size, nullptr)) {
+        buff.resize(size);
+        if (PeekNamedPipe(pipe, &buff[0], size, &count, nullptr, nullptr)) {
+            return String((char *)buff.ptr(), size);
```

Comment:
Same here, with the extremely inconsistent Windows terminal encodings, it's most likely won't properly work even with `String::utf8((char *)buff.ptr(), size)`.

Reply:
you answered my previous question, will do!

Reply:
@bruvzg wanted to point out this code has been tested outside of Godot with here: https://github.com/time-killer-games/xproc and it does work sending UTF-8 strings to standard input, at least when doing reading and writing to stdin within C++, though I can't remember whether it actually works in command prompt or not. I'll give it a test.

Reply:
It depends on system locale and `chcp` value (if it's not set to UTF-8, outputting UTF-8 directly might break), so pretty hard to test.

Reply:
Didn't notice that last thing you said about PeekConsoleInputW until just now. My bad!

Reply:
After reading microsoft documentation, it really seems that PeekConsoleInputW only works with console applications, so any user who wants to read from stdin from a gui application won't be able to do so from this approach. I think I'm going to stick with PeekNamedPipe until we can find something which works with files sent directly to stdin.

Reply:
@bruvzg I don't think PeekConsole is actually needed, as long as the user does the things you pointed out with their system locale and `chcp`. I added the ReadFile code and verified in my `xproc` repository that indeed fixed the issue with `executable < file`. Nice catch! :D